Mathlib/Topology/ContinuousMap/CompactlySupported.lean

```diff
@@ -541,6 +540,99 @@ noncomputable def nnrealPart (f : C_c(α, ℝ)) : C_c(α, ℝ≥0) where
+                                       apply HasCompactSupport.comp_left f.hasCompactSupport' rfl⟩
+
+@[simp]
+lemma toReal_apply (f : C_c(α, ℝ≥0)) (x : α) :
+    f.toReal x = f x := rfl
+
+lemma eq_nnrealPart_neg_nnrealPart (f : C_c(α, ℝ)) :
+    f = (nnrealPart f).toReal - (nnrealPart (-f)).toReal := by
+  ext x
+  simp only [coe_sub, Pi.sub_apply, toReal_apply, nnrealPart_apply, Real.coe_toNNReal', coe_neg,
+    Pi.neg_apply, max_zero_sub_max_neg_zero_eq_self]
+
+/-- The compactly supported continuous `ℝ≥0`-valued function as a compactly supported `ℝ`-valued
+function. -/
+noncomputable def LinearMap.toReal : C_c(α, ℝ≥0) →ₗ[ℝ≥0] C_c(α, ℝ) where
+  toFun := fun f => ⟨ContinuousMap.coeNNRealReal.comp f.1,
+                    by
+                    simp only [ContinuousMap.toFun_eq_coe, ContinuousMap.coe_comp,
+                      ContinuousMap.coeNNRealReal_apply,
+                      CompactlySupportedContinuousMap.coe_toContinuousMap]
+                    exact HasCompactSupport.comp_left f.hasCompactSupport' (by rfl)⟩
+  map_add' f g := by
+    ext x
+    simp only [CompactlySupportedContinuousMap.coe_mk, ContinuousMap.comp_apply,
+      CompactlySupportedContinuousMap.coe_toContinuousMap, CompactlySupportedContinuousMap.coe_add,
+      Pi.add_apply, ContinuousMap.coeNNRealReal_apply, NNReal.coe_add, ContinuousMap.coe_comp,
+      Function.comp_apply]
+  map_smul' a f := by
+    ext x
+    simp only [CompactlySupportedContinuousMap.coe_mk, ContinuousMap.comp_apply,
+      CompactlySupportedContinuousMap.coe_toContinuousMap, CompactlySupportedContinuousMap.coe_smul,
+      Pi.smul_apply, smul_eq_mul, ContinuousMap.coeNNRealReal_apply,
+      RingHom.id_apply, ContinuousMap.coe_comp, Function.comp_apply]
+    rw [NNReal.smul_def, smul_eq_mul]
+    simp only [NNReal.coe_mul]
+
+@[simp]
+lemma LinearMap.toReal_apply (f : C_c(α, ℝ≥0)) (x : α) :
+    LinearMap.toReal f x = (f x).toReal := rfl
+
+lemma LinearMap.coe_toReal (f : C_c(α, ℝ≥0)) :
+    LinearMap.toReal f = f.toReal := rfl
+
+/-- For a positive linear functional `Λ : C_c(α, ℝ) → ℝ`, define a `ℝ≥0`-linear map. -/
+noncomputable def toNNRealLinear {Λ : C_c(α, ℝ) →ₗ[ℝ] ℝ} (hΛ : ∀ f, 0 ≤ f.1 → 0 ≤ Λ f) :
+    C_c(α, ℝ≥0) →ₗ[ℝ≥0] ℝ≥0 where
+  toFun := fun f => ⟨Λ (LinearMap.toReal f),
+                    by
+                    apply hΛ (LinearMap.toReal f)
+                    intro x
+                    simp only [ContinuousMap.toFun_eq_coe, ContinuousMap.zero_apply,
+                      coe_toContinuousMap, LinearMap.toReal_apply, zero_le_coe]
+                    ⟩
+  map_add' f g := by
+    simp only [map_add]
+    exact rfl
+  map_smul' a f := by
+    simp only [map_smul, LinearMap.map_smul_of_tower, RingHom.id_apply, smul_eq_mul]
+    exact rfl
+
+@[simp]
+lemma toNNRealLinear_apply {Λ : C_c(α, ℝ) →ₗ[ℝ] ℝ} (hΛ : ∀ f, 0 ≤ f.1 → 0 ≤ Λ f) (f : C_c(α, ℝ≥0)) :
+    toNNRealLinear hΛ f = (Λ (toReal f)) := rfl
```

Comment:
You tend to add extraneous brackets in your expressions. Could you please go through your PRs and make sure you don't have any such?
```suggestion
    toNNRealLinear hΛ f = Λ (toReal f) := rfl
```